asreviewcontrib/datatools/dedup.py

```diff
@@ -1,38 +1,44 @@
 import argparse
 
+import numpy as np
 import pandas as pd
 
-from asreview.data import load_data
 
+def dedup(asdata, pid='doi'):
+    initial_length = len(asdata.df)
 
-def dedup(input_path, output_path=None):
-    # read data in ASReview data object
-    asdata = load_data(input_path)
+    if pid in asdata.df.columns:
+        # replace NaN values with empty string to prevent astype(str) from creating literal 'NaN' string
+        asdata.df[pid].fillna('', inplace=True)
+
+        # convert to string to support non-string types, strip whitespaces and replace empty strings with NaN values
+        s_pid = asdata.df[pid].astype(str).str.strip().replace("", np.nan)
```

Comment:
I doublechecked, but None==None returns true while np.nan==np.nan returns false. Also with '.duplicated()' it will see Nones as dups unfortunately

Reply:
But `pd.isnull(None)` and `pd.isnull(np.nan)` is both True. 

I don't think you have to cast, strip, or replace. 

```python
>>> s = pd.Series([1,2,2,3,3, None, None])
>>> s
0    1.0
1    2.0
2    2.0
3    3.0
4    3.0
5    NaN
6    NaN
dtype: float64
>>> ~(s.duplicated() & s.notnull())
0     True
1     True
2    False
3     True
4    False
5     True
6     True
dtype: bool
>>> s[~(s.duplicated() & s.notnull())]
0    1.0
1    2.0
3    3.0
5    NaN
6    NaN
dtype: float64
```

Reply:
You are right, None can be used instead of np.nan 👍 However removing cast, strip and replace from `s_pid = asdata.df[pid].astype(str).str.strip().replace("", None)` causes empty string and strings with whitespaces to be counted as duplicates, so I think that piece of code should remain! Currently updating the PR

Reply:
```python 
>>> import pandas as pd
>>> s = pd.Series(["a", "", "c"])
>>> s.dtype
dtype('O')
>>> from pandas.api.types import is_string_dtype
>>> is_string_dtype(s)
True
```